src/ULog.ts

```diff
@@ -241,29 +243,57 @@ export class ULog {
       return;
     }
 
-    for (let i = range[0]; i <= range[1]; i++) {
-      const [_timestamp, offset, msgId] = timeIndex[i]!;
+    if (opts.reverse === true) {
+      for (let i = range[1]; i >= range[0]; i--) {
+        const [_timestamp, offset, msgId] = timeIndex[i]!;
+
+        if (includeLogs && msgId === LogMessageId) {
+          reader.seekTo(offset);
+          const msg = await this.#readParsedMessage(reader);
+          if (msg) {
+            yield msg;
+            continue;
+          }
+        }
+
+        // If there are message ids specified, then only yield if the locator msgId matches
+        if (msgIds != undefined) {
+          if (msgId == undefined || !msgIds.has(msgId)) {
+            continue;
+          }
+        }
 
-      if (includeLogs && msgId === LogMessageId) {
         reader.seekTo(offset);
         const msg = await this.#readParsedMessage(reader);
         if (msg) {
           yield msg;
-          continue;
         }
       }
```

Comment:
it's a bit of a shame we have to duplicate all this – I see that you can't trivially put it into a reusable function because of the `yield` – but perhaps you could make a reusable function that returns an _optional_ message and then `if (msg) yield msg;`?

Alternatively, you could set `inc = opts.reverse ? -1 : 1` and do `i += inc` (also swapping start and end values)?

Reply:
To swap start/end I would need to also swap the if ending condition...